engine/src/main/scala/cromwell/engine/workflow/package.scala

```diff
@@ -0,0 +1,40 @@
+package cromwell.engine
+
+import akka.actor.{ActorSystem, ActorRef}
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.services.MetadataServiceActor.{MetadataServiceAction, PutMetadataAction, MetadataServiceResponse, MetadataEvent}
+
+import scala.concurrent.{Future, ExecutionContext}
+
+package object workflow {
+  // Sent when there is a failure to push to the metadata service
+  private[workflow] case class MetadataServiceFailureResponse(reason: Throwable, action: Option[MetadataServiceAction] = None)
+
+  object WorkflowMetadataKeys {
+    val Name = "workflowName"
+    val Id = "id"
+    val SubmissionTime = "submission"
+    val StartTime = "start"
+    val EndTime = "end"
+    val Status = "status"
+    val Outputs = "outputs"
+    val Inputs = "inputs"
+    val Calls = "calls"
+  }
+
+  implicit class EnhancedServiceRegistry(val serviceRegistryActor: ActorRef) extends StrictLogging {
+    def publishToMetadataService(message: MetadataEvent)(implicit actorSystem: ActorSystem, ec: ExecutionContext): Future[MetadataServiceResponse] = {
+      import cromwell.util.PromiseActor._
+
+      // Since this will always publish, we expect the callee to just provide an event,
+      // and we wrap it into a `PutMetadataAction` before sending it to the MetadataService
+      serviceRegistryActor.askNoTimeout(PutMetadataAction(message)).mapTo[MetadataServiceResponse].recover {
```

Comment:
Also if you really want to use an ask inside of another actor's receive loop, you might want to check out this library: https://github.com/jducoeur/Requester